internal/pkg/siderolink/manager.go

```diff
@@ -654,11 +656,31 @@ func (manager *Manager) Provision(ctx context.Context, req *pb.ProvisionRequest)
 			return nil, err
 		}
 
-		dirty = true
+		updated = true
 	}
 
-	if dirty {
+	if created || updated {
 		if err = manager.wgHandler.PeerEvent(ctx, spec, false); err != nil {
+			if !created {
```

Comment:
this feels still like a partial fix to me.... ? as updated will try only once same way

I wonder if we should somehow tie the resource create/update with the PeerEvent error?

I don't see the full picture though

Reply:
I would agree with Andrey here.
Probably the right way would be to move create/update logic after we successfully do `PeerEvent`, that will naturally fix all issues with this code.

So:
1. get the link
2. if it exists or the key hasn't changed: return.
3. update the spec in memory, send peer event.
4. do create or update call with the created spec.